flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -127,13 +128,29 @@
     private boolean forceAvro = false;
     private long autoWatermarkInterval = 200;
 
+    // ---------- statebackend related configurations ------------------------------
     /**
      * Interval in milliseconds for sending latency tracking marks from the sources to the sinks.
      */
     private long latencyTrackingInterval = MetricOptions.LATENCY_INTERVAL.defaultValue();
 
     private boolean isLatencyTrackingConfigured = false;
 
+    /** Interval in milliseconds to perform periodic materialization. */
+    private long periodicMaterializeInterval =
+            StateBackendOptions.PERIODIC_MATERIALIZATION_INTERVAL.defaultValue();
+
+    /** Interval in milliseconds for initial delay of periodic materialization. */
+    private long periodicMaterializeInitDelay =
+            StateBackendOptions.PERIODIC_MATERIALIZATION_INIT_DELAY.defaultValue();
+
+    /** Max allowed number of failures */
+    private int materializationMaxAllowedFailures =
+            StateBackendOptions.MATERIALIZATION_MAX_ALLOWED_FAILURES.defaultValue();
+
+    /** Flag to enable periodic materialization */
+    private boolean isPeriodicMaterializationEnabled = false;
```

Comment:
What are the use caseswhen this flag is false?

Reply:
I was thinking for testing purposes. 

Not for the unit test, but for the randomization test, when we only want to test changelog itself, not including materialization.

WDYT?

Reply:
I'm not sure whether we should randomize it; but it seems we can achieve it with a very big materialization interval, can't we?

Reply:
Not to randomize materialization. It provides a way that only uses changelog  
OR changelog + materialization.

For testing purposes (at least for now), I think it is quite important.

Using very big intervals could also be a solution, but it depends on the implementation of how materialization is implemented, which as you commented (very practical).

Reply:
Yes, I got what you meant regarding "changelog OR changelog + materialization" and that's why I proposed an alternative (big intervals).
It doesn't introduce new config option (and new code) and that's why I think it's preferrable.

Reply:
Resolved by removing the flag.